### PR TITLE
[ANI-101] Farmer report API

### PR DESCRIPTION
**Pull request by AldwinVergara**
### Proposed changes
-Added the asController function in CreateFarmerProfile
-Added path
-Added proxy swagger info (I'm not yet really sure what to insert there)

### Checklist

-   [x] I updated my feature branch with the `master` branch.
-   [x] I have ran `composer lint` and it is successful.
-   [x] I have ran `composer test` and it is successful.

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -19,3 +21,6 @@
     Route::post('/login', LoginFarmer::class)->name('login');
     Route::post('/logout', LogoutFarmer::class)->name('logout')->middleware('auth:api');
 });
+
+Route::post('/create.farmer.report', CreateFarmerReport::class);
```
Comment (jg-rivera) [inline routes/api.php:25]: Add `->middleware('auth:api')` to create farmer report API route

File: routes/api.php [hunk 2]
```diff
@@ -19,3 +21,6 @@
...
+Route::post('/delete.farmer.report', DeleteFarmerReport::class);
```
Comment (jg-rivera) [inline routes/api.php:26]: Remove delete farmer report API route. Farmer dashboard will not call this.

File: routes/api.php (hunk 1 above)
Comment (jg-rivera) [inline routes/api.php:25]: Revise `/create.farmer.report` API route. Do not use dots in URLs. Use `/farmer-reports` instead. Check out this [reference](https://usercontent.one/wp/www.kennethlange.com/wp-content/uploads/2020/04/customer_rest_api.png) in how to structure API routes based on the REST standard.

File: app/Actions/FarmerReport/DeleteFarmerReport.php [hunk 3]
```diff
@@ -27,4 +27,8 @@ public function asOrchidAction($model, ?Request $request)
 
         return redirect()->route('platform.farmer-reports');
     }
+
+    public function asController()
```
Comment (jg-rivera) [inline app/Actions/FarmerReport/DeleteFarmerReport.php:31]: Delete `asController` function. No need to call delete farmer report from API.

File: app/Actions/FarmerReport/CreateFarmerReport.php [hunk 4]
```diff
@@ -32,6 +33,36 @@ public function asOrchidAction($model, ?Request $request)
         return redirect()->route('platform.farmer-reports');
     }
 
+    //PROXYYYY PALANG TOH HAHAHAH di ko pa alam ano ilalagay hehehe
```
Comment (jg-rivera) [inline app/Actions/FarmerReport/CreateFarmerReport.php:36]: Make the Swagger documentation more appropriate. Change the values inside each property. For example, define the required parameters to create a farmer report in the `@OA\JsonContent` part of the comment.

File: app/Actions/FarmerReport/CreateFarmerReport (ORIG).php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace App\Actions\FarmerReport;
+
+use App\Models\FarmerReport\FarmerReport;
+use App\Traits\AsOrchidAction;
+use Illuminate\Http\Request;
+use Lorisleiva\Actions\Concerns\AsAction;
+use Orchid\Support\Facades\Toast;
+
+class CreateFarmerReport
```
Comment (jg-rivera) [inline app/Actions/FarmerReport/CreateFarmerReport (ORIG).php:11]: Delete this file when you are done.

**Review by jg-rivera — CHANGES_REQUESTED** [review]
See comments.

File: routes/api.php (hunk 1 above)
Comment (AldwinVergara) [reply routes/api.php:25]: Changed the URL.

File: app/Actions/FarmerReport/DeleteFarmerReport.php (hunk 3 above)
Comment (AldwinVergara) [reply app/Actions/FarmerReport/DeleteFarmerReport.php:31]: Deleted the asController function in the delete farmer report.

File: app/Actions/FarmerReport/CreateFarmerReport.php (hunk 4 above)
Comment (AldwinVergara) [reply app/Actions/FarmerReport/CreateFarmerReport.php:36]: Updated the Swagger documentation.

Comment (jg-rivera) [reply app/Actions/FarmerReport/CreateFarmerReport.php:36]: CI is not passing. Check the error log [here](https://github.com/Radical-Doubting/anikultura-backend/runs/4997322562?check_suite_focus=true).

Comment (AldwinVergara) [reply app/Actions/FarmerReport/CreateFarmerReport.php]: Fixed and updated the issure regarding multiple @OA\Property.

**Review by jg-rivera — APPROVED** [review]

**Comment by jg-rivera** [comment]
Good job!

**Pull request merged**